### PR TITLE
Support tags in template and stack config

### PR DESCRIPTION
**Pull request by herebebogans**
As per #22 an initial implementation of tags support for stack creation.

### CODE REVIEW

**Comment by herebebogans** [comment]
I've made stack_config INHERITABLE_PROPERTIES now override template_config .. I added a dependency on hashie for deep_merge as it's used in the specs in a similar way.

File: lib/bora/stack.rb [hunk 1]
```diff
@@ -316,7 +316,12 @@ def run_cfndsl(template_file, params, pretty_json)
     end
 
     def cfn_options_from_stack_config
-      valid_options = ["capabilities"]
+      valid_options = %w(capabilities tags)
+
+      # Expand Tags to key, value pairs
+      if @stack_config[:tags]
+        @stack_config[:tags] = @stack_config[:tags].collect { |k, v| { key: k, value: v } }
+      end
```
Comment (ampedandwired) [inline lib/bora/stack.rb]: Mutating `@stack_config` doesn't feel quite right here. Creates up side effects, for example what if this method happened to be called twice? Maybe assign the result of the "select" on the next line below to a temporary var, and expand the tags on that?

Comment (herebebogans) [inline lib/bora/stack.rb]: OK - Will do.

**Comment by herebebogans** [comment]
Updated!

**Comment by ampedandwired** [comment]
Released as 1.6.0

**Comment by herebebogans** [comment]
Sorry - pry as a development dependency slipped in.

**Pull request merged**